emigration.lua

```diff
@@ -112,42 +187,17 @@ local function event_loop()
     end
 end
 
-dfhack.onStateChange[GLOBAL_KEY] = function(sc)
-    if sc == SC_MAP_UNLOADED then
-        enabled = false
-        return
-    end
-
-    if sc ~= SC_MAP_LOADED or df.global.gamemode ~= df.game_mode.DWARF then
-        return
+dfhack.onStateChange.loadEmigration = function(code)
+    if code==SC_MAP_LOADED then
+        if enabled then
+            print("Emigration enabled.")
+            event_loop()
+        else
+            print("Emigration disabled.")
+        end
     end
-
-    local persisted_data = json.decode(persist.GlobalTable[GLOBAL_KEY] or '')
-    enabled = (persisted_data or {enabled=false})['enabled']
-    event_loop()
-end
-
-if dfhack_flags.module then
-    return
 end
 
-if df.global.gamemode ~= df.game_mode.DWARF or not dfhack.isMapLoaded() then
-    dfhack.printerr('emigration needs a loaded fortress map to work')
-    return
+if dfhack.isMapLoaded() then
+    dfhack.onStateChange.loadEmigration(SC_MAP_LOADED)
 end
-
-local args = {...}
-if dfhack_flags and dfhack_flags.enable then
-    args = {dfhack_flags.enable_state and 'enable' or 'disable'}
-end
-
-if args[1] == "enable" then
-    enabled = true
-elseif args[1] == "disable" then
-    enabled = false
-else
-    return
-end
-
-event_loop()
-persist_state()
```

Comment:
this part also looks like it was overwritten with old code

Reply:
Sorry!